tests/rptest/tests/pandaproxy_test.py

```diff
@@ -907,56 +914,311 @@ def topics_appeared():
                    err_msg="Timeout waiting for topics to appear.")
 
 
-class PandaProxyBasicAuthTest(RedpandaTest):
+class PandaProxyBasicAuthTest(PandaProxyEndpoints):
     password = 'simple'
     algorithm = 'SCRAM-SHA-256'
 
     def __init__(self, context):
-        extra_rp_conf = dict(auto_create_topics_enabled=False, )
-
         security = SecurityConfig()
         security.enable_sasl = True
         security.endpoint_authn_method = 'sasl'
         security.pp_authn_method = 'http_basic'
 
-        super(PandaProxyBasicAuthTest,
-              self).__init__(context,
-                             num_brokers=3,
-                             enable_pp=True,
-                             security=security,
-                             extra_rp_conf=extra_rp_conf)
+        super(PandaProxyBasicAuthTest, self).__init__(context,
+                                                      security=security)
 
-        self.admin = Admin(self.redpanda)
+    @cluster(num_nodes=3)
+    def test_get_brokers(self):
+        # Regular user without authz priviledges
+        # should fail
+        res = self._get_brokers(auth=('red', 'panda')).json()
+        assert res['error_code'] == 40101
 
-        http.client.HTTPConnection.debuglevel = 1
-        http.client.print = lambda *args: self.logger.debug(" ".join(args))
+        super_username, super_password, _ = self.redpanda.SUPERUSER_CREDENTIALS
+        brokers_raw = self._get_brokers(auth=(super_username, super_password))
+        brokers = brokers_raw.json()['brokers']
```

Comment:
please remember to split up your commits so that each commit does one thing (within reason). so far in this commit there is renaming of the test class, chagning _get_brokers, changing _get_topics, kwargs is being plumbed (but im not sure why).

also, it sort of looks like some of the new code you are adding (e.g. test_get_brokers) could be added at the end of the file to reduce some of the noise in the diff. for example, what i highlighted here where it looks like a new test is being added, but it's being mixed in with other line addition/deletions.

Reply:
kwargs is so we don't have to keep adding new parameters to request helpers.